pkg/csi/service/vanilla/controller.go

```diff
@@ -269,13 +271,29 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 				"Error:%+v", vcconfig.Host, err)
 		}
 		if !isvSANFileServicesSupported {
-			vsanFileServiceNotSupported = true
-			break
+			allVCentersSupportvSANFileService = false
+		}
+
+		isCnsTransactionSupported, err := c.managers.VcenterManager.IsCnsTransactionSupported(ctx,
+			vcconfig.Host)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to verify if CNS transaction is supported or not for vCenter: %s. "+
+				"Error:%+v", vcconfig.Host, err)
+		}
+		if !isCnsTransactionSupported {
+			allVCentersSupportCnsTransaction = false
```

Comment:
On a multi vCenter setup, if one of the VC is upgraded and has transaction support enabled, we need to allow  creating volume using transaction support on it.

Reply:
For this case if any of the vCenter is less than 9.1.0, then transaction support will remain disabled.

Reply:
As discussed offline: we will keep the transaction support disabled even if one of the multiple vc is not upgraded.